controllers/drplacementcontrol_controller.go

```diff
@@ -1213,24 +1223,272 @@ func (r *DRPlacementControlReconciler) updateResourceCondition(
+//     requiring further processing of vrg for a status decision
+func ProcessVRGDataReadyAsPrimary(drpc *rmn.DRPlacementControl,
+	vrg *rmn.VolumeReplicationGroup,
+	clusterName string,
+) bool {
+	updated := true
+
+	condition := meta.FindStatusCondition(vrg.Status.Conditions, VRGConditionTypeDataReady)
+
+	if condition != nil && condition.Status == metav1.ConditionTrue && condition.ObservedGeneration == vrg.Generation {
+		return !updated
+	}
+
+	if condition == nil || condition.ObservedGeneration != vrg.Generation || condition.Status == metav1.ConditionUnknown {
+		SetDRPCStatusCondition(&drpc.Status.Conditions, rmn.ConditionProtected, drpc.Generation,
+			metav1.ConditionFalse,
+			rmn.ReasonProtectedProgressing,
+			fmt.Sprintf("VolumeReplicationGroup on cluster %s is progressing on initializing data protection",
+				clusterName))
+
+		return updated
+	}
+
+	// condition.Status == metav1.ConditionFalse
+	// TODO: Expand and provide better context in message based on reason for VRGConditionTypeDataReady being false
+	SetDRPCStatusCondition(&drpc.Status.Conditions, rmn.ConditionProtected, drpc.Generation,
+		metav1.ConditionFalse,
+		rmn.ReasonProtectedError,
+		fmt.Sprintf("VolumeReplicationGroup on cluster %s has errors initializing data protection, "+
+			"retrying till %s condition is met",
+			clusterName, VRGConditionTypeDataReady))
+
+	return updated
+}
+
+// ProcessVRGDataProtectedAsPrimary is a helper function used by ProcessDataReadyAndProtectedAsPrimary
+//   - Returns a bool that is true if status was updated, and false otherwise,
+//     requiring further processing of vrg for a status decision
+func ProcessVRGDataProtectedAsPrimary(drpc *rmn.DRPlacementControl,
+	vrg *rmn.VolumeReplicationGroup,
+	clusterName string,
+) bool {
+	updated := true
+
+	condition := meta.FindStatusCondition(vrg.Status.Conditions, VRGConditionTypeDataProtected)
+
+	if condition != nil && condition.ObservedGeneration == vrg.Generation {
+		if condition.Reason == VRGConditionReasonReplicating && condition.Status == metav1.ConditionFalse &&
+			vrg.Spec.Async != nil {
+			return !updated
 		}
 
-		drpc.Status.ResourceConditions.ResourceMeta.ProtectedPVCs = protectedPVCs
-		drpc.Status.LastGroupSyncTime = vrg.Status.LastGroupSyncTime
+		if condition.Status == metav1.ConditionTrue && vrg.Spec.Sync != nil {
+			return !updated
+		}
+	}
+
+	if condition == nil || condition.ObservedGeneration != vrg.Generation || condition.Status == metav1.ConditionUnknown {
+		SetDRPCStatusCondition(&drpc.Status.Conditions, rmn.ConditionProtected, drpc.Generation,
+			metav1.ConditionFalse,
+			rmn.ReasonProtectedProgressing,
+			fmt.Sprintf("VolumeReplicationGroup on cluster %s is progressing on establishing data synchronization",
+				clusterName))
 
-		r.setLastSyncTimeMetric(syncmetric, vrg.Status.LastGroupSyncTime, log)
+		return updated
 	}
+
+	if (condition.Status == metav1.ConditionTrue && vrg.Spec.Async != nil) ||
```

Comment:
Shouldn't the condition be false?
`if (condition.Status == metav1.ConditionFalse && vrg.Spec.Async != nil) ||`

Reply:
Please recheck, should be addressed in the revised commits.